doc/release-notes/releases.md

```diff
@@ -0,0 +1,8 @@
+# sourmash release notes
```

Comment:
Do we want to point to the github release notes in here? Dunno if we want to do it for each version, but maybe could add some text saying, "minor etc releases are documented <here>"

Reply:
punted to https://github.com/dib-lab/sourmash/issues/1052